src/charm.py

```diff
@@ -1547,8 +1547,14 @@ def update_config(self, is_creating_backup: bool = False) -> bool:
             logger.warning("Early exit update_config: Cannot connect to Postgresql")
             return False
 
+        # Use config value if set, calculate otherwise
+        if self.config.experimental_max_connections:
+            max_connections = self.config.experimental_max_connections
+        else:
+            max_connections = max(4 * os.cpu_count(), 100)
+
```

Comment:
Doing this only here would trigger a restart on deploy with `experimental_max_connections`.  I can also inject it in the config file, by changing the internal library, but I would rather keep it KISS.